pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -179,6 +179,9 @@ func (bfs *Service) oidDownloader() {
 
 	var err error
 	if bfs.cfg.SkipIndexFilesSearch {
+		//Depends on https://github.com/nspcc-dev/neo-go/issues/3645.
+		//We need to search with EQ filter to prevent not full results.
+		bfs.cfg.OIDBatchSize = 1
```

Comment:
And you need to change `fetchOIDsBySearch` accordingly to use EQUAL instead of GE/LE if a single object is requested.